src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.InitOnlySetters)]
+    public class InitOnlyMemberTests : CompilingTestBase
+    {
+        // Spec: https://github.com/dotnet/csharplang/blob/master/proposals/init.md
+
+        // PROTOTYPE(init-only): test allowed from 'with' expression
+        // PROTOTYPE(init-only): public API, confirm behavior of IsReadOnly and IsInitOnly
+
+        // PROTOTYPE(init-only): open issues:
+        // PROTOTYPE(init-only): queue discussion on init methods (`init void Init()`) and collection initializers (`init void Add()`)
+
+        // PROTOTYPE(init-only): test dynamic scenario
+        // PROTOTYPE(init-only): test whether reflection use property despite modreq?
+        // PROTOTYPE(init-only): test behavior of old compiler with modreq. For example VB
+        // PROTOTYPE(init-only): test with ambiguous IsExternalInit types
+
+        [Fact]
+        public void TestCSharp8()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init; }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (4,35): error CS8652: The feature 'init-only setters' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public string Property { get; init; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "init").WithArguments("init-only setters").WithLocation(4, 35)
+                );
+        }
+
+        [Fact]
+        public void TestInitNotModifier()
+        {
+            string source = @"
+public class C
+{
+    public string Property { get; init set; }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,40): error CS8180: { or ; or => expected
+                //     public string Property { get; init set; }
+                Diagnostic(ErrorCode.ERR_SemiOrLBraceOrArrowExpected, "set").WithLocation(4, 40),
+                // (4,40): error CS1007: Property accessor already defined
+                //     public string Property { get; init set; }
+                Diagnostic(ErrorCode.ERR_DuplicateAccessor, "set").WithLocation(4, 40)
+                );
+        }
+
+        [Fact]
+        public void TestWithDuplicateAccessor()
```

Comment:
>TestWithDuplicateAccessor [](start = 20, length = 25)

I think it would be useful to assert symbol model. How many method besides constructor are we going to end up with in the type, what is their kind, are they InitOnly, etc.